src/bootstrap/src/bin/rustc.rs

```diff
@@ -150,7 +150,7 @@ fn main() {
         {
             cmd.arg("-Ztls-model=initial-exec");
         }
-    } else if std::env::var("MIRI").is_err() {
```

Comment:
This was added [here](https://github.com/rust-lang/rust/pull/118071#issuecomment-1820971141) but at least in local testing is not needed any more. Let's see what CI says.

Cc @Urgau

Reply:
If it's not needed anymore, that probably mean that Miri is respecting the `RUSTFLAGS` set by bootstrap (or always passing a target). It's probably not worth checking if there are any `--cfg bootstrap` or `--check-cfg` args being passed, since we will quickly see if CI fails.

Good to see this hack disappear.

Reply:
Miri always passes a target, but that has always been the case. Nothing recently changed wrt RUSTFLAGS either.

But maybe somehow this interacts with the other changes in this PR.